lib/controllers/getTarball.js

```diff
@@ -4,27 +4,43 @@ function getTarball(respond, route, unpm) {
   var version = route.params.version
   var name = route.params.name
   var started = false
+  var getTarballAsync = unpm.Package.getTarballAsync
+  var getTarballSync = unpm.Package.getTarball
 
-  var tarballStream = unpm.Package.getTarball(name, version)
+  var tarballStream = null
 
-  tarballStream.on('error', function(err) {
-    if(started) {
-      return unpm.log.error(err)
-    }
+  if (typeof getTarballSync === 'function') {
+    // Use synchronous behavior for fetching tarball. We're assuming it's coming from disk
+    tarballStream = getTarballSync(name, version)
+    doneFnc()
```

Comment:
it is generally bad practice to conditionally either call a function synchronously (as done here) or asynchronously (as done in the else clause).  This is what leads to things like the current test failures (where `doneFnc` is not yet defined), but works fine when calling it asynchronously.

Instead you can do something like:  `process.nextTick(doneFnc)`